src/Components/Payment/PaymentForm.tsx

```diff
@@ -0,0 +1,288 @@
+import { Box, Button, Flex, Join, Serif, Spacer, Theme } from "@artsy/palette"
+import { PaymentFormCreateCreditCardMutation } from "__generated__/PaymentFormCreateCreditCardMutation.graphql"
+import {
+  Address,
+  AddressErrors,
+  AddressForm,
+  AddressTouched,
+  emptyAddress,
+} from "Apps/Order/Components/AddressForm"
+import { CreditCardInput } from "Apps/Order/Components/CreditCardInput"
+import { validatePresence } from "Apps/Order/Components/Validators"
+import { ErrorModal } from "Components/Modal/ErrorModal"
+import React, { Component } from "react"
+import { commitMutation, graphql, RelayProp } from "react-relay"
+import { injectStripe, ReactStripeElements } from "react-stripe-elements"
+import { ConnectionHandler } from "relay-runtime"
+import { Responsive } from "Utils/Responsive"
+
+export interface PaymentFormProps
+  extends ReactStripeElements.InjectedStripeProps {
+  relay?: RelayProp
+  me: any
+}
+
+interface PaymentFormState {
+  address: Address
+  hideBillingAddress: boolean
+  error: stripe.Error
+  isCommittingMutation: boolean
+  isErrorModalOpen: boolean
+  errorModalMessage: string
+  addressErrors: AddressErrors
+  addressTouched: AddressTouched
+}
+
+class PaymentForm extends Component<PaymentFormProps, PaymentFormState> {
+  private cardElement
+
+  state = {
+    address: { ...emptyAddress, country: "US" },
+    hideBillingAddress: true,
+    error: null,
+    isCommittingMutation: false,
+    isErrorModalOpen: false,
+    errorModalMessage: null,
+    addressErrors: {},
+    addressTouched: {},
+  }
+
+  get touchedAddress() {
+    return {
+      name: true,
+      country: true,
+      postalCode: true,
+      addressLine1: true,
+      addressLine2: true,
+      city: true,
+      region: true,
+      phoneNumber: true,
+    }
+  }
+
+  onSubmit = () => {
+    const billingAddress = this.getSelectedBillingAddress()
+    const { me } = this.props
+    this.setState({ isCommittingMutation: true }, () => {
+      const errors = this.validateAddress(this.state.address)
+      if (Object.keys(errors).filter(key => errors[key]).length > 0) {
```

Comment:
Would it make sense to maybe encompass this bit within `validateAddress` so that it can be checked like 

`if (errors.hasErrors)` 

(or something)